src/interceptors/http.ts

```diff
@@ -26,8 +26,16 @@ export class HttpInterceptors {
    *```
    */
   public static setDefaultHeaders(): RequestInterceptor {
+    const DEFAULT_HEADERS = new Map([['Content-Type', 'application/json']]);
+
     return ({ request }) => {
-      request.headers.set('Content-Type', 'application/json');
+      for (const [key, value] of DEFAULT_HEADERS.entries()) {
```

Comment:
we might want to compare it case-insensitive or else if someone adds `'content-type' -> 'text/plain'` it will still be included

Reply:
The `Headers.prototype.has` and `Headers.prototype.get` already handle case-insensitive matching. I've added a test to make sure the behavior stays consistent across node versions

Reply:
(Headers names are meant to be case-insensitive)

Reply:
ahhh, oops, sorry!